File: ExLab/Java/Lab2/Ex1.java
/**
 *
 * autor: Pedro Ruiz
 * data: 9/3/2018
 * código matéria: CC3642
 * 
 * propósito: criar um programa com uma classe Funcionario e exibir seu salário
 */
package Ex1;

class Funcionario {  //cria classe funcionario

    String nome;       //atributos da classe
    String sobrenome;   
    double salario;     

    public void setSobrenome(String novoSobrenome) { //função set
        sobrenome = novoSobrenome;
    }

    public void setSalario(double novoSalario) { 
        if (novoSalario < 0) {
            salario = 0;          //função set com verificação de valor
        } else {
            salario = novoSalario;
        }
    }

    public void setNome(String novoNome) {  //função set
        nome = novoNome;
    }

    public String getNome() {  //função get
        return nome;
    }

    public String getSobrenome() {  //função get
        return sobrenome;
    }

    public double getSalario() {     //função get
        return salario;
    }
    
    public void imprime(){ //função que imprime nome, sobrenome e salario
        System.out.println("nome: " + getNome());
        System.out.println("sobrenome: " + getSobrenome());
        System.out.println("salario anual: " + getSalario() * 12);
    }

}


public class Ex1 {

    
    
     

    // classe main roda o programa
    public static void main(String[] args) {

        Funcionario f = new Funcionario(); //instancia o objeto
        f.setNome("Adriano"); //da nome ao atributo nome do objeto
        f.setSobrenome("Lopes"); //da sobrenome ao atributo nome do objeto
        f.setSalario(30000.5);  //da salario ao atributo nome do objeto

        Funcionario g = new Funcionario();  //instancia o objeto
        g.setNome("Gilmar"); //da nome ao atributo nome do objeto
        g.setSobrenome("Mendes"); //da sobrenome ao atributo nome do objeto
        g.setSalario(999999); //da salario ao atributo nome do objeto
        
        f.imprime(); //imprime
        g.imprime(); //imprime

        f.setSalario(f.getSalario() * 1.1); //aumento de 10% no salario
        g.setSalario(g.getSalario() * 1.1); //aumento de 10% no salario
        
        f.imprime(); //imprime com salario alterado
        g.imprime(); //imprime com salario alterado
        
    }

}

File: ExLab/Cpp/Lab2/Ex1.cpp
/*
*autor: Pedro Henrique Nardoni Ruiz
*
*data: 9/3/2018
*
*codigo da materia: CC3642
*
*explicação: criar um programa com uma classe Funcionario e exibir seu salário
*
*/





#include <iostream>
#include <string>

using namespace std;

class Funcionario { //cria classe funcionario
public: //metodos da classe
	void setSobrenome(string newSobrenome) { //funçao set sobrenome
		sobrenome = newSobrenome;
	}

	void setNome(string newNome) { //funçao set nome
		nome = newNome;
	}

	void setSalario(double newSalario) { //funçao set salario com verificaçao                                      de entrada
		if (newSalario < 0) {
			salario = 0;
		} else {
			salario = newSalario;
		}
	}

	string getSobrenome() { //funçao get sobrenome
		return sobrenome;
	}

	string getNome() { //funçao get nome
		return nome;
	}

	double getSalario() { //funçao get salario
		return salario;
	}

	void imprime() { //funçao que imprime os atributos
		cout << "Nome:" << getNome() << endl;
		cout << "Sobrenome:" << getSobrenome() << endl;
		cout << "Salario anual:" << getSalario() * 12 << endl;
	}

private:  //atributos da classe
	string nome;
	string sobrenome;
	double salario;
};

int main() { //funçao main roda o programa
	Funcionario f1; //instanciando primeiro objeto

	f1.setNome("Pedro"); //dando nome
	f1.setSobrenome("Ruiz"); //dando sobrenome
	f1.setSalario(10); //dando salario

	Funcionario f2; //instanciando segundo objeto
	f2.setNome("Gilmar"); //dando nome
	f2.setSobrenome("Mendes"); //dando sobrenome
	f2.setSalario(50); //dando salario

	f1.imprime(); //imprimindo
	f2.imprime(); //imprimindo

	f1.setSalario(f1.getSalario() * 1.1); //aumentando 10% no salario
	f2.setSalario(f2.getSalario() * 1.1); //aumentando 10% no salario

	f1.imprime(); //imprimindo
	f2.imprime(); //imprimindo
}